### PR TITLE
kaidan: 0.8.0 -> 0.9.1

### LINKED ISSUE #274285: Build failure: kaidan
### Steps To Reproduce

1. build *kaidan*

### Build log

```
error: builder for '/nix/store/b96r47bbiq0wiid1d97nhfd0pv9lhrj1-kaidan-0.8.0.drv' failed with exit code 2;
       last 10 log lines:
       > [ 71%] Building CXX object CMakeFiles/kaidan.dir/src/CredentialsGenerator.cpp.o
       > make[2]: *** [CMakeFiles/kaidan.dir/build.make:629: CMakeFiles/kaidan.dir/src/QrCodeDecoder.cpp.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > /build/source/src/QrCodeGenerator.cpp: In static member function 'static QImage QrCodeGenerator::generateQrCode(int, const QString&)':
       > /build/source/src/QrCodeGenerator.cpp:77:71: error: 'QR_CODE' is not a member of 'ZXing::BarcodeFormat'
       >    77 |                 ZXing::MultiFormatWriter writer(ZXing::BarcodeFormat::QR_CODE);
       >       |                                                                       ^~~~~~~
       > make[2]: *** [CMakeFiles/kaidan.dir/build.make:643: CMakeFiles/kaidan.dir/src/QrCodeGenerator.cpp.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:138: CMakeFiles/kaidan.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/b96r47bbiq0wiid1d97nhfd0pv9lhrj1-kaidan-0.8.0.drv'.
```

### Notify maintainers

@astro 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[ultra@titan:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.4, NixOS, 24.05 (Uakari), 24.05.20231204.2c7f3c0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `""`
 - channels(ultra): `""`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by astro**
## Description of changes

Fixes #274285 

Kaidan supports OMEMO now!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/kaidan/default.nix [hunk 1]
```diff
@@ -18,14 +21,14 @@
...
-  version = "0.8.0";
+  version = "0.9.1";
 
   src = fetchFromGitLab {
     domain = "invent.kde.org";
     owner = "network";
     repo = pname;
     rev = "v${version}";
-    sha256 = "070njci5zyzahmz3nqyp660chxnqx1mxp31w17syfllvrw403qmg";
+    sha256 = "sha256-F5GhN9hAF2e8b0T3peUnLk8CVd+nq4YR8k52x6ZOoLM=";
```
Comment (kirillrdy) [inline pkgs/applications/networking/instant-messengers/kaidan/default.nix:31]:
```suggestion
    hash = "sha256-F5GhN9hAF2e8b0T3peUnLk8CVd+nq4YR8k52x6ZOoLM=";
```

File: pkgs/by-name/li/libomemo-c/package.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "libomemo-c";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "dino";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-GvHMp0FWoApbYLMhKfNxSBel1xxWWF3TZ4lnkLvu2s4=";
```
Comment (kirillrdy) [inline pkgs/by-name/li/libomemo-c/package.nix:16]:
```suggestion
    hash = "sha256-GvHMp0FWoApbYLMhKfNxSBel1xxWWF3TZ4lnkLvu2s4=";
```

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 274823` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>kaidan</li>
    <li>libomemo-c</li>
    <li>qxmpp</li>
  </ul>
</details>

File: pkgs/by-name/li/libomemo-c/package.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildsInputs = [ openssl ];
+  cmakeFlags = [ "-DBUILD_SHARED_LIBS=ON" ];
+
+  meta = with lib; {
+    description = "Fork of libsignal-protocol-c adding support for OMEMO XEP-0384 0.5.0+";
+    homepage = "https://github.com/dino/libomemo-c";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/by-name/li/libomemo-c/package.nix:26]:
```suggestion
    license = licenses.gpl3Only;
```

File: pkgs/by-name/li/libomemo-c/package.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+, openssl
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libomemo-c";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "dino";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/by-name/li/libomemo-c/package.nix:14]:
```suggestion
    repo = "libomemo-c";
```

**Review by SuperSandro2000 — APPROVED** [review]
nit

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by astro** [comment]
@nixpkgs-merge-bot merge

**Comment by kirillrdy** [comment]
@astro i guess merge-bot didnt work because those conditions we not met

```
To ensure security and a focused utility, the bot adheres to specific limitations:

    Only compatible with packages located under pkgs/by-name/*.

    For now we only allow pull requests done by r-ryantm to be merged

```

**Pull request merged**